libs/MeshKernelApi/src/MeshKernel.cpp

```diff
@@ -988,21 +988,42 @@ namespace meshkernelapi
             }
             else
             {
+                throw meshkernel::AlgorithmError("Invalid MakeGridParameters");
+            }
+        }
+        catch (...)
+        {
+            exitCode = HandleExceptions(std::current_exception());
+        }
+        return exitCode;
+    }
 
-                if (!meshkernel::IsEqual(makeGridParameters.angle, 0.0))
-                {
-                    throw meshkernel::AlgorithmError("When generating an uniform grid on an defined extension, the grid angle must be equal to 0");
-                }
+    MKERNEL_API int mkernel_mesh2d_make_uniform_on_extension(int meshKernelId,
```

Comment:
Check if mkernel_curvilinear_make_uniform_on_extension can be used here to duplication

Reply:
i removed the duplication by introducing a new function. Note that in the case of mesh2d_make_uniform_... we do not want to add a curvilinear grid in the meshkernel state, so calling mkernel_curvilinear_make_uniform_on_extension inside mesh2d_make_uniform_... will not work